### DISCUSSION TITLE
Genesis import, the "sample" parameter

### DISCUSSION THREAD

**Discussion by werin99** [ocelot-collab/ocelot#238] [Q&A]
I import .fld files from Genesis 1.3 v4, but the divergence of the field varies with the value on "sample" in the Genesis simulation.
While the size of the field follows very well sqrt(beta*eps) the divergence is neither close to a gaussian light beam, sqrt(eps/beta) or a combination thereof. 
Any good ideas?

**Comment by ZeugAusHH** [comment]
Hi @werin99,

could you please provide us with a small OCELOT script demonstrating the reported behavior?
Which version of OCELOT are you using?

Best regards,
Christoph

**Reply by werin99** [reply]
Hello, thanks for looking!
Here is the code:

# Read an output file from Genesis 1.3 v4
dfl_file=read_dfl4('z:\SRC\src240422nc.70.fld.h5') # large file, sample=100, beta =15
print(dfl_file.fld.shape)
plot_dfl(dfl_file, fig_name='dfl w spectrum', domains='kf')

dfl_file2=read_dfl4('z:\SRC\src240419nc.70.fld.h5') # very large file, many slices, sample=10, beta=15
print(dfl_file2.fld.shape)
plot_dfl(dfl_file2, fig_name='dfl w spectrum', domains='kf')

And the output:
![fig1](https://github.com/ocelot-collab/ocelot/assets/45360647/328888a0-8822-42bb-9940-105dd5e4c0c2)
![fig2](https://github.com/ocelot-collab/ocelot/assets/45360647/92f6f28f-3927-4e7f-b17f-ccd3f67f2560)

If I see correctly I use Ocelot 20.11.2
Cheers
- Sverker

**Reply by ZeugAusHH** [reply]
Dear Sverker,

many thanks for sharing these materials. Unfortunately, I could not reproduce the issue, I get the same sigma_x' in both cases.

I noticed you're using a quite old release of OCELOT. Could you try this with a more recent issue? I did my tests with git commit id 47773a8, this is the status of branch `dev_gen` of the OCELOT code as of 2024-04-29. Maybe updating OCELOT solves the issue already.

Also, you could try with a "perfect" field generated in "GENESIS 1.3" (this is what I used for my attempt to reproduce). For instance as in the attached input file.

Hope that helps.

Best
Christoph

[t.in.txt](https://github.com/ocelot-collab/ocelot/files/15148640/t.in.txt)
[t.lat.txt](https://github.com/ocelot-collab/ocelot/files/15148669/t.lat.txt)
[show_fld.py.txt](https://github.com/ocelot-collab/ocelot/files/15148657/show_fld.py.txt)

**Reply by werin99** [reply]
Thanks Christoph
With your input files it is correct. What is clearer in them is that the spectral bandwidth plotted and, I assume, used for the divergence cacluation is different in the two cases. I can now see that it is the case in my results above too. With this limitation in BW the divergence agrees to the Theta-E_ph plots. Those plots also show a stronger "folding in" at higher photon energies when "sample" is large.
I now assume my problem is to broaden the BW. Then I will probably also get rid of the "folding-in". Is there another way than reducing the "sample" in the original Genesis simulation? 
Cheers
Sverker
